### ISSUE TITLE
DNS/TLS for Workbench

### ISSUE THREAD

**Issue by craig-willis** [terraref/computing-pipeline#290]
The Workbench service now at http://www.terraref.ndslabs.org needs a wildcard TLS cert.  The ndslabs.org domain is a temporary testing domain and is not intended to host services.

Completion criteria:
* [x] Get access to terraref.org DNS or possibly transfer ownership to NCSA
* [x] Add *.workbench.terraref.org wildcard DNS to terraref.org
* [x] Wildcard TLS certificate for workbench.terraref.org

**Comment by craig-willis** [comment]
Have access to account. Opened ticket with NCSA security to see if they will issue the cert or if we need to move ownership.

**Comment by craig-willis** [comment]
For NCSA to provide the certificate - you will need to move the domain registration and coordinate with NCSA NetEng 

You can also get wild card domain certs from a commercial provider. 
https://www.digicert.com/wildcard-ssl-certificates.htm

**Comment by craig-willis** [comment]
Opened ticket with NCSA+NetEng about moving the domain.

**Comment by craig-willis** [comment]
From neteng: 

"Yes the domain can be transferred. I think it costs usually around $12 and some
change (basically the cost of an additional year of renewal). The admin contact
is [ ], so he'll need to provide NCSA with the Authorization
Code which you can retrieve from inside your Registrar portal. NCSA will then
make the request for transfer with that Code, and the Admin contact  will
get an email which he'll need to ok. Usually takes about a week.

What point of complex, looks like your using ARPLUS.NET has your DNS provider.
When you make the switch to NCSA we'll need to have all of your DNS records
ported over to our system to prevent much disruption. So that will be something
else you will need to provide, probably before we start the transfer.

**Comment by craig-willis** [comment]
From neteng:

"I've created terraref.org on NCSA DNS servers, and setup an A record to point
to the IP you provided. You can log into your registrar now and change the name
servers to NCSA DNS server. So replace: ns1.aplus.net ns2.aplus.net with dns1.ncsa.illinois.edu
dns2.ncsa.illinois.edu."

I've done this via aplus.net interface.

**Comment by craig-willis** [comment]
Received message from NCSA neteng that transfer failed because it was not approved.   Will re-initiate transfer.

**Comment by craig-willis** [comment]
Re-initiated transfer to NCSA.  This will likely be done next week.

**Comment by craig-willis** [comment]
Transfer complete!  Next steps are to request the new workbench domain and TLS certificates. 

@dlebauer My plan is to use workbench.terraref.org, unless you have another suggestion?

**Comment by dlebauer** [comment]
@craig-willis I think workbench.terraref.org is fine but can we have terraref.ndslabs.org redirect here at least for a few months so we can update links in documentation?

At some point it would be nice to point, e.g., bety to traits.terraref.org and clowder to sensors.terraref.org - is that difficult to do?

**Comment by craig-willis** [comment]
DNS request for *.workbench.terraref.org submitted. Will address the bety/traits and clowder/sensors requests in a separate ticket. (It should be possible, but in discussions with Rob there are configuration changes for Clowder/Bety).

**Comment by craig-willis** [comment]
Workbench DNS entry in place.  Will need to update the instance configuration and add the redirect from terraref.ndslabs.org

**Comment by craig-willis** [comment]
Requested wildcard SSL cert.

**Comment by craig-willis** [comment]
Update from NCSA CA group -- apparently they have to submit to InCommon to allow NCSA to sign certs for terraref.org. This is still being processed.

**Comment by craig-willis** [comment]
NCSA is now able to sign SSL certificates for terraref.org. In the future, we can request new certificates by emailing help+ca@ncsa.illinois.edu.  I have the wildcard certs for Workbench now and need to install and add a redirect entry from the old URL.

**Comment by craig-willis** [comment]
Need to discuss strategy for storing keys (i.e., Last Pass)

**Comment by robkooper** [comment]
Use lastpass for sharing, that is what NCSA supports. I already have an account there for some information.

**Comment by craig-willis** [comment]
@dlebauer 
I need to schedule downtime for workbench to install the SSL certificates.  I'll need <1 hour, but will need to shutdown your running applications.  I'd like to get this done this week, if possible.  Let me know if there are any good/bad times to do this.

**Comment by dlebauer** [comment]
I don't see any bad times this week. If anyone is using it please send an email warning in advance.

**Comment by craig-willis** [comment]
@dlebauer You are the only person running with active containers at the moment.

I'll likely take it down tonight to take care of several maintenance tasks: install SSL certs, mount the ROGER users directory, resize the storage volume.
